### PR TITLE
refactor: zetaclient metrics

### PR DESCRIPTION
**Pull request by skosito**
# Description

- use gauge and counter vectors with labels, instead of separate gauge/counter per public key or chain

- simplify how gauges and counters are initialized and used around zetaclient, metrics not passed around as argument anymore and nesting in chain client and chain metrics structs removed

- cleanup code

i also removed `zetaclient_Outbound_tx_sign_count`, it seems like it is not updated

tested locally, with these changes:

```
zetaclient_hotkey_burn_rate 5.3958333333333e+13
# HELP zetaclient_pending_txs_total Number of pending transactions per chain
# TYPE zetaclient_pending_txs_total gauge
zetaclient_pending_txs_total{chain="btc_regtest"} 0
zetaclient_pending_txs_total{chain="goerli_localnet"} 0
# HELP zetaclient_rpc_getBlockByNumber_count Count of getLogs per chain
# TYPE zetaclient_rpc_getBlockByNumber_count counter
zetaclient_rpc_getBlockByNumber_count{chain="goerli_localnet"} 20
# HELP zetaclient_rpc_getFilterLogs_count Count of getLogs per chain
# TYPE zetaclient_rpc_getFilterLogs_count counter
zetaclient_rpc_getFilterLogs_count{chain="goerli_localnet"} 38
# HELP zetaclient_tss_node_blame_count Tss node blame counter per pubkey
# TYPE zetaclient_tss_node_blame_count counter
zetaclient_tss_node_blame_count{pubkey="zetapub1addwnpepqftghjat0y44p9c0e7lnejva5gsxeqd49cpe9sl9glem858j883xg7wqphq"} 1
zetaclient_tss_node_blame_count{pubkey="zetapub1addwnpepqtgur2ze59nax2pwlefqq0ndks6axwxpm3lkcsusvgv42wdz0v9fk5vses9"} 1
```

before changes:
```
# HELP zetaclient_Outbound_tx_sign_count number of Outbound tx signed
# TYPE zetaclient_Outbound_tx_sign_count counter
zetaclient_Outbound_tx_sign_count 0
# HELP zetaclient_hotkey_burn_rate Fee burn rate of the hotkey
# TYPE zetaclient_hotkey_burn_rate gauge
zetaclient_hotkey_burn_rate 2.7421875e+13
# HELP zetaclient_pending_txs_btc_regtest Number of pending transactions
# TYPE zetaclient_pending_txs_btc_regtest gauge
zetaclient_pending_txs_btc_regtest 0
# HELP zetaclient_pending_txs_goerli_localnet Number of pending transactions
# TYPE zetaclient_pending_txs_goerli_localnet gauge
zetaclient_pending_txs_goerli_localnet 0
# HELP zetaclient_rpc_getBlockByNumber_count_goerli_localnet Number of getBlockByNumber
# TYPE zetaclient_rpc_getBlockByNumber_count_goerli_localnet counter
zetaclient_rpc_getBlockByNumber_count_goerli_localnet 9
# HELP zetaclient_rpc_getFilterLogs_count_goerli_localnet Number of getLogs
# TYPE zetaclient_rpc_getFilterLogs_count_goerli_localnet counter
zetaclient_rpc_getFilterLogs_count_goerli_localnet 16
# HELP zetaclient_zetapub1addwnpepqded5mp3uap6440s9k4xn5x7se3zthkhndu3ep4y8pc5g8660np87atzrcg_tss tss node blame counter
# TYPE zetaclient_zetapub1addwnpepqded5mp3uap6440s9k4xn5x7se3zthkhndu3ep4y8pc5g8660np87atzrcg_tss counter
zetaclient_zetapub1addwnpepqded5mp3uap6440s9k4xn5x7se3zthkhndu3ep4y8pc5g8660np87atzrcg_tss 0
# HELP zetaclient_zetapub1addwnpepqg0tsrtyg5kucs2e3nt0ehgttadjc9rfwjd8fq6gr3uv7cnqadhd6x3932l_tss tss node blame counter
# TYPE zetaclient_zetapub1addwnpepqg0tsrtyg5kucs2e3nt0ehgttadjc9rfwjd8fq6gr3uv7cnqadhd6x3932l_tss counter
zetaclient_zetapub1addwnpepqg0tsrtyg5kucs2e3nt0ehgttadjc9rfwjd8fq6gr3uv7cnqadhd6x3932l_tss 0
```

Closes: #1431
#1769 

## Type of change

Refactor

# How Has This Been Tested?

Not yet

- [ ] Tested CCTX in localnet
- [x] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [x] I have added unit tests that prove my fix feature works

### CODE REVIEW

**Review by kevinssgh — COMMENTED** [review]
Looks like the correct direction, will review once its ready.

File: changelog.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 * [1630](https://github.com/zeta-chain/node/pull/1630) added password prompts for hotkey and tss keyshare in zetaclient
   Starting zetaclient now requires two passwords to be input; one for the hotkey and another for the tss key-share.
 * [1760](https://github.com/zeta-chain/node/pull/1760) - Make staking keeper private in crosschain module
+* [1783](https://github.com/zeta-chain/node/pull/1783) - refactor zetaclient metrics naming and structure
```
Comment (lumtis) [inline changelog.md:24]: Need to be moved under `Unreleased`
Will check if we can have the CI checking for this

**Review by lumtis — APPROVED** [review]
This looks good to me.
Will need a review from @zeta-chain/devops, so I'm adding ci label

File: zetaclient/zetacore_observer.go [hunk 2]
```diff
@@ -7,25 +7,21 @@ import (
...
 	"github.com/zeta-chain/zetacore/x/crosschain/types"
 	"github.com/zeta-chain/zetacore/zetaclient/config"
-	"github.com/zeta-chain/zetacore/zetaclient/metrics"
 )
 
 const (
-	MaxLookaheadNonce   = 120
-	OutboundTxSignCount = "zetaclient_Outbound_tx_sign_count"
-	HotKeyBurnRate      = "zetaclient_hotkey_burn_rate"
+	MaxLookaheadNonce = 120
```
Comment (CharlieMc0) [inline zetaclient/zetacore_observer.go:24]: I thought this was set via chain params instead of being hard coded?

**Review by CharlieMc0 — APPROVED** [review]
LGTM. Just had a question about the lookahead nonce being hardcoded but I'll follow up outside of this PR

**Pull request merged**